### PR TITLE
memory leak fixes

### PR DESCRIPTION
**Pull request by yashlyft**
/cc @theatrus

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -522,7 +523,7 @@ int prepare_prefixes(statsite_config *config)
         assert(res != -1);
         config->prefixes_final[i] = final_prefix;
     }
-    return 0;
+    return final_prefix;
```
Comment (theatrus) [inline src/config.c]: We only capture one final prefix? We should probably enumerate the config on shutdown instead.

File: src/config.c [hunk 2]
```diff
@@ -766,6 +767,22 @@ void free_config(statsite_config* config) {
...
+        free(config->log_level);
+    }
+    if (config->pid_file != NULL) {
+        free(config->pid_file);
+    }
+    if (config->prefixes[COUNTER] != NULL) {
+        free(config->prefixes[COUNTER]);
+    }
+
+    for (int i = 0; i < METRIC_TYPES; i++) {
```
Comment (theatrus) [inline src/config.c]: Like here :)

File: src/sink_http.c [hunk 3]
```diff
@@ -216,7 +216,7 @@ static int serialize_metrics(struct http_sink* sink, metrics* m, void* data) {
 
     int post_len = 0;
     char* post_data = strbuf_get(post_buf, &post_len);
-    strbuf_free(post_buf, false);
+    strbuf_free(post_buf, true);
```
Comment (theatrus) [inline src/sink_http.c]: Don't want this - we would free the char* array in the strbuf here, and pass a freed memory to the lifoq

Comment (theatrus) [inline src/sink_http.c]: The leak is likely later in the code, somewhere after lifoq dequeue and in the curl error handling path

File: src/config.c (hunk 1 above)
Comment (yashlyft) [inline src/config.c]: yeah we dont quite need this, residual. Let me delete and go back return int

File: src/sink_http.c [hunk 4]
```diff
@@ -387,8 +388,9 @@ static void* http_worker(void* arg) {
 
         if (should_authenticate && s->oauth_bearer == NULL) {
             if (!oauth2_get_token(httpconfig, s)) {
-                if (lifoq_push(s->queue, data, data_size, true, true))
+                if (lifoq_push(s->queue, data, data_size, true, true)) {
                     syslog(LOG_ERR, "HTTP: dropped data due to queue full of closed");
+                }
```
Comment (theatrus) [inline src/sink_http.c]: Should probably free here too

**Comment by theatrus** [comment]
+1, one more spot to free

**Comment by yashlyft** [comment]
:rocket:

**Pull request merged**